File: Algorithms/Implementation/Strange Counter.php
<?php

/*
 * Complete the 'strangeCounter' function below.
 *
 * The function is expected to return a LONG_INTEGER.
 * The function accepts LONG_INTEGER t as parameter.
 */

function strangeCounter($t) {
    # Write your code here
    $a = 3;
    while ($t > $a) {
        $t -= $a;
        $a *= 2;
    }

    return ($a - $t + 1);
}

$fptr = fopen(getenv("OUTPUT_PATH"), "w");

$t = intval(trim(fgets(STDIN)));

$result = strangeCounter($t);

fwrite($fptr, $result . "\n");

fclose($fptr);

File: Algorithms/Implementation/Strange Counter.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;

class Result
{

    /*
     * Complete the 'strangeCounter' function below.
     *
     * The function is expected to return a LONG_INTEGER.
     * The function accepts LONG_INTEGER t as parameter.
     */

    public static long strangeCounter(long t)
    {
        long a = 3;
        while (t > a) {
            t -= a;
            a *= 2;
        }

        return (a - t + 1);
    }
}

class Solution
{
    public static void Main(string[] args)
    {
        TextWriter textWriter = new StreamWriter(@System.Environment.GetEnvironmentVariable("OUTPUT_PATH"), true);

        long t = Convert.ToInt64(Console.ReadLine().Trim());

        long result = Result.strangeCounter(t);

        textWriter.WriteLine(result);

        textWriter.Flush();
        textWriter.Close();
    }
}
